assessments-service/src/test/java/com/transformuk/hee/tis/assessment/service/service/impl/AssessmentServiceImplTest.java

```diff
@@ -534,6 +536,32 @@ public void advancedSearchShouldSearchUsingSpecifications() {
     assertThat("Unexpected first name.", assessmentListDto.getFirstName(), is("firstName2"));
   }
 
+  @Test
+  public void advancedSearchShouldSearchWithColumnFilters() {
+    String reviewDateString = "2021-01-01";
+    Assessment assessment1 = new Assessment();
+    assessment1.setId(1L);
+    assessment1.setTraineeId(TRAINEE_ID);
+    assessment1.setReviewDate(LocalDate.parse(reviewDateString));
+
+    ColumnFilter columnFilter = new ColumnFilter("reviewDate",
+        Lists.newArrayList(reviewDateString));
+    List<ColumnFilter> columnFilters = Lists.newArrayList(columnFilter);
+    Pageable pageable = new PageRequest(0, 20);
+
+    List<Assessment> foundAssessments = Lists.newArrayList(assessment1);
+    Page<Assessment> pagedFoundAssessments = new PageImpl<>(foundAssessments);
+
+    when(assessmentRepositoryMock.findAll(specificationsArgumentCaptor.capture(), eq(pageable)))
```

Comment:
I can't see the assertion that the right value has been captured.

Reply:
> 3\. Give yourself credit for a _feature_ & increase the service pom by a minor version rather than patch.

Always up for giving myself credit :), but in terms of the service, I had thought really the only changes is a bugfix on handling columnFilter date fields correctly! Are you sure its feature-worthy?

Reply:
Thanks, a bit of copy-pasta there (on the argumentCaptor). I spent a silly amount of time a couple of days ago trying in a unit test to figure out how to interrogate specification lambdas after they have been created, e.g. to check that they are what they should be. I failed, which is why I've just gotten rid of the captor now :). But if you have any tips on unit testing the details of the lambdas I'd love to know for next time!